app/src/components/Scheduling/CalendarView.js

```diff
@@ -0,0 +1,85 @@
+import { React, useState, useEffect } from 'react';
+import DayPicker, { DateUtils } from 'react-day-picker';
+import styles from './CalendarView.css';
+import Header from './Header';
+import { Button } from '../../store/index';
+import { useStyles } from './CalendarViewStyles';
+
+export default function CalendarView() {
+  const classes = useStyles();
+  const [selectedDays, handleDayChange] = useState([]);
+  const [buttonDisabled, setButtonDisabled] = useState(true);
+
```

Comment:
Would be a little cleaner to define `handleButtonNext` outside of the `useEffect` and then call it inside the `useEffect`. So something like 
 ```
 const handleButtonNext = () => {
    if (selectedDays.length > 0) {
      setButtonDisabled(false);
    } else {
      setButtonDisabled(true);
    }
  };

useEffect(() => {
    handleButtonNext();
}, [selectedDays])
```

Reply:
I moved the `handleButtonNext` outside, but now I'm getting an eslint warning:
<img width="1182" alt="Screen Shot 2021-04-28 at 3 55 24 PM" src="https://user-images.githubusercontent.com/17840013/116482634-6326f880-a83a-11eb-8494-4a1161548c05.png">

And if I do add `handleButtonNext` to the dependency array in `useEffect`, the `handleButtonNext` function gives me this warning:

`The 'handleButtonNext' function makes the dependencies of useEffect Hook (at line 23) change on every render. Move it inside the useEffect callback. Alternatively, wrap the definition of 'handleButtonNext' in its own useCallback() Hook.`

What would be the best course to handle this?

Reply:
Hmm that's strange. I just pulled down your code, and moved the `handleButtonNext` function outside the useEffect hook so it looks like this and I'm not getting any errors or warnings. If you play around with it more and it's still being weird, then we can take a look at together.

```
  const handleButtonNext = () => {
    if (selectedDays.length > 0) {
      setButtonDisabled(false);
    } else {
      setButtonDisabled(true);
    }
  };
  useEffect(() => {
    handleButtonNext();
  }, [selectedDays]);
```